spec/javascripts/models/beta/story_spec.js

```diff
@@ -246,107 +246,179 @@ describe('Story model', function () {
+          describe(`when new status is ${status[item]}`, () => {
+            it(`change status to ${status[item]}`, () => {
+              const story = { _editing: { storyType: noFeatureType } };
+              const newAttributes = { state: status[item] }
 
-        const changedStory = Story.editStory(story, newAttributes);
+              const changedStory = Story.editStory(story, newAttributes);
 
-        expect(changedStory).toEqual({
-          _editing: {
-            labels: [],
-            storyType: 'feature',
-            estimate,
-            state: 'unstarted',
-            _isDirty: true
-          }
+              expect(changedStory._editing.state).toEqual(status[item]);
+            });
+          });
         });
-      });
 
-      it(`change story state to unscheduled when estimate is null and sets _isDirty to true`, () => {
-        const story = { _editing: { storyType: 'feature', estimate, state: 'unstarted' } };
-        const newAttributes = { estimate: '' };
+        describe(`when story status is ${UNSCHEDULED}`, () => {
+          describe(`when new story type is feature`, () => {
+            it('change estimate to "" and storyType to feature', () => {
+              const story = { _editing: { 
+                storyType: noFeatureType, 
+                estimate: '', 
+                state: UNSCHEDULED } 
+              };
+              const newAttributes = { storyType: FEATURE }
+    
+              const changedStory = Story.editStory(story, newAttributes);
+    
+              expect(changedStory._editing.storyType).toEqual(FEATURE);
+              expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
 
-        const changedStory = Story.editStory(story, newAttributes);
+        describe(`when story status is ${UNSTARTED}`, () => {
+          describe(`when new story type is feature`, () => {
+            it('change story estimate to 1 and storyType to feature', () => {
+              const story = { _editing: { 
+                  storyType: noFeatureType,
+                  estimate: 2,
+                  state: UNSTARTED
+                } 
+              };
+              const newAttributes = { storyType: FEATURE }
+
+              const changedStory = Story.editStory(story, newAttributes);
+
+              expect(changedStory._editing.storyType).toEqual(FEATURE);
+              expect(changedStory._editing.state).toEqual(UNSTARTED);
+              expect(changedStory._editing.estimate).toEqual(1);
+            });
+          });
+        });
 
-        expect(changedStory).toEqual({
-          _editing: {
-            labels: [],
-            storyType: 'feature',
-            estimate: '',
-            state: 'unscheduled',
-            _isDirty: true
-          }
+        describe(`when state is ${UNSTARTED}`, () => {
+          describe(`when new state is ${UNSCHEDULED}`, () => {
```

Comment:
You can apply this style https://github.com/Codeminer42/cm42-central/pull/576/files#diff-e7cad8ffcafac32387e195a4957fe897R257-R273 here.